### PR TITLE
DDF-2183 STS LDAP login bundle uses the configured username attribute

### PR DESCRIPTION
**Pull request by jrnorth**
#### What does this PR do?
- Updates the LDAP login bundle so it will actually use the configured username attribute
- Updates the LDAP Claims Handler and LDAP Login configuration documentation

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@codice/security 
@bdeining 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@stustison

#### How should this be tested?
Set up the embedded LDAP server (or some other LDAP server) with a username attribute other than "uid". Start the LDAP login bundle and configure it to use that username attribute. Verify that you can log in via LDAP.

#### What are the relevant tickets?
[DDF-2183](https://codice.atlassian.net/browse/DDF-2183)

#### Checklist:
- [x] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by stustison** [comment]
👍

File: platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapService.java [hunk 1]
```diff
@@ -54,10 +53,8 @@ public LdapService(final BundleContext context) {
      * @param newModule that will replace a module or be added to the list of modules.
      */
     public synchronized void update(Module newModule) {
-        modules = modules.stream()
-                .filter(m -> !m.getName()
-                        .equals(newModule.getName()))
-                .collect(Collectors.toList());
+        modules.removeIf(m -> m.getName()
```
Comment (coyotesqrl) [inline platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapService.java]: 1. I'm not generally a fan of mutable collections, though the fact that this method is synchronized (and, I assume, all the other methods that modify the `modules` list are as well) helps.
2. However, you're risking a future problem if someone changes the implementation class of `modules` to a Collection whose iterator doesn't support `remove`. It might be best to change its declaration to `ArrayList` to be specific. And, while you're there, you could also make it `final` since you're no longer replacing it but mutating it.

**Comment by coyotesqrl** [comment]
👍

File: platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapService.java (hunk 1 above)
Comment (jrnorth) [inline platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapService.java]: Looking at this again, it seems like a map would be more appropriate than a list. But this also seems like it's outside the scope of this ticket.

**Comment by bdeining** [comment]
:+1:

**Comment by jlcsmith** [comment]
built and tested successfully 👍
